.github/workflows/ci.yaml

```diff
@@ -267,26 +253,24 @@ jobs:
         uses: 'google-github-actions/auth@ef5d53e30bbcd8d0836f4288f5e50ff3e086997d'
         with:
           credentials_json: '${{ secrets.GCP_KEY }}'
-      - name: Run Integration Tests
-        shell: bash -l {0}
-        run: |
-          flit install -s
-          pytest -n auto --cov=datajudge --cov-report=xml --cov-append --backend=bigquery tests/integration
-      - name: Generate code coverage report
-        uses: codecov/codecov-action@v3.1.1
+      - name: Run tests
+        uses: ./.github/actions/pytest
         with:
-          file: ./coverage.xml
-
+          backend: bigquery
+          args: tests/integration
 
-  linux-integration_tests-impala-column-pt1:
-    name: "Linux - integration tests - Python ${{ matrix.PYTHON_VERSION }} - Impala - pt1"
+  impala-column:
+    if: ${{ contains(github.event.pull_request.labels.*.name, 'impala') || contains(github.event.pull_request.labels.*.name, 'ready') }}
+    name: "Python ${{ matrix.PYTHON_VERSION }} - SA ${{ matrix.SA_VERSION }} - Impala - pt1"
     runs-on: ubuntu-20.04
     env:
       CI: True
     strategy:
       fail-fast: false
       matrix:
         PYTHON_VERSION: [ '3.8']
+        SA_VERSION: ["<=2.0", ">=2.0"]
+        PYTEST_ARG: ["tests/integration/test_column_capitalization.py", "tests/integration/test_data_source.py", "tests/integration/test_integration.py -k row", "tests/integration/test_integration.py -k uniques", "tests/integration/test_integration.py -k date", "tests/integration/test_integration.py -k varchar", "tests/integration/test_integration.py -k numeric"]
```

Comment:
this might be controversial but I replaced the 7 impala jobs (they all looked the same :D) with a matrix argument that is passed to the pytest action which
just looked shorter, behaviour should be the same.

Reply:
Great idea!